pkg/utils/dangerous_secret_names.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+var DANGEROUS_SECRET_NAMES = []string{
+	"PROMPT_COMMAND",
+	"LD_PRELOAD",
+	"LD_LIBRARY_PATH",
+	"WINDIR",
+	"USERPROFILE",
+	"DYLD_INSERT_LIBRARIES",
+}
+
+type DangerousSecretNameError struct {
+	dangerousNames []string
+	currentConfig  string
+}
+
+func (e *DangerousSecretNameError) Error() string {
+	return fmt.Sprintf("the following dangerous secret names have been found in %s config:\n%s", e.currentConfig, strings.Join(e.dangerousNames, "\n"))
+}
+
+func CheckForDangerousSecretNames(secrets map[string]string) *DangerousSecretNameError {
```

Comment:
I think this would be more appropriate as a controller function. Utils are meant to be more generic